scylla/src/transport/connection.rs

```diff
@@ -47,7 +47,7 @@ use std::{
 };
```

Comment:
commit: `iterator: adjust to the new deserialization framework `

> Unfortunately, due to the limitations of the Stream trait, currently the TypedRowIterator cannot deserialize types that need to borrow from the serialized frame contents.

I've played around with this, by removing `RowT: 'static` bounds and trying to create a `TypedRowStream` over some borrowed type (tried with `&str` and my custom wrapper over `&str`). I think that compiler error messages are not really insightful.. But this is only my opinion.

Funny thing is that compiler returns two different messages. For simply `&str`, with the following snippet:
```rust
let typed_iter = raw_iter.into_typed::<(&str,)>().unwrap();
let mut typed_stream = typed_iter.into_stream();
let s = typed_stream.next().await;
```
what I get is:
```
error: implementation of `DeserializeRow` is not general enough
    --> scylla/src/transport/connection.rs:1270:17
     |
1270 |         let s = typed_stream.next().await;
     |                 ^^^^^^^^^^^^^^^^^^^ implementation of `DeserializeRow` is not general enough
     |
     = note: `(&str,)` must implement `DeserializeRow<'0>`, for any lifetime `'0`...
     = note: ...but it actually implements `DeserializeRow<'1>`, for some specific lifetime `'1`

error: implementation of `DeserializeRow` is not general enough
    --> scylla/src/transport/connection.rs:1270:37
     |
1270 |         let s = typed_stream.next().await;
     |                                     ^^^^^ implementation of `DeserializeRow` is not general enough
     |
     = note: `(&str,)` must implement `DeserializeRow<'0>`, for any lifetime `'0`...
     = note: ...but it actually implements `DeserializeRow<'1>`, for some specific lifetime `'1`
```
, while for the snippet with my custom wrapper:
```rust
struct StrWrapper<'a> {
    s: &'a str,
}

// implementations of DeserializeRow and DeserializeValue for StrWrapper

...


let typed_iter = raw_iter.into_typed::<(StrWrapper,)>().unwrap();
let mut typed_stream = typed_iter.into_stream();
let s = typed_stream.next().await;
```
I get:
```
error[E0599]: no method named `next` found for struct `TypedRowStream` in the current scope
    --> scylla/src/transport/connection.rs:1270:30
     |
1270 |         let s = typed_stream.next().await;
     |                              ^^^^ method not found in `TypedRowStream<(StrWrapper<'_>,)>`
     |
    ::: scylla/src/transport/iterator.rs:1068:1
     |
1068 | pub struct TypedRowStream<RowT> {
     | ------------------------------- method `next` not found for this struct
     |
     = help: items from traits can only be used if the trait is implemented and in scope
     = note: the following traits define an item `next`, perhaps you need to implement one of them:
             candidate #1: `Iterator`
             candidate #2: `StreamExt`
help: trait `StreamExt` which provides `next` is implemented but not in scope; perhaps you want to import it
     |
1    + use futures::StreamExt;
     |

```

Reply:
In the second case (your wrapper), you must have forgotten to `use futures::StreamExt as _`, correct?

Reply:
No. Previously, I did not include the warning about unused import which appeared as well.
```
error[E0599]: no method named `next` found for struct `TypedRowStream` in the current scope
    --> scylla/src/transport/connection.rs:1238:20
     |
1238 |         let s = ts.next().await;
     |                    ^^^^ method not found in `TypedRowStream<(StrWrapper<'_>,)>`
     |
    ::: scylla/src/transport/iterator.rs:1065:1
     |
1065 | pub struct TypedRowStream<RowT: 'static> {
     | ---------------------------------------- method `next` not found for this struct
     |
     = help: items from traits can only be used if the trait is implemented and in scope
     = note: the following traits define an item `next`, perhaps you need to implement one of them:
             candidate #1: `Iterator`
             candidate #2: `StreamExt`
help: trait `StreamExt` which provides `next` is implemented but not in scope; perhaps you want to import it
     |
1    + use futures::StreamExt;
     |

warning: unused import: `futures::StreamExt`
    --> scylla/src/transport/connection.rs:1235:13
     |
1235 |         use futures::StreamExt as _;
     |             ^^^^^^^^^^^^^^^^^^
     |
     = note: `#[warn(unused_imports)]` on by default

```